airflow-core/src/airflow/serialization/serialized_objects.py

```diff
@@ -1787,12 +1788,140 @@ def to_dict(cls, var: Any) -> dict:
+                    dag_dict["timetable"] = {
+                        "__var": {},
+                        "__type": "airflow.timetables.simple.OnceTimetable",
+                    }
+                elif sched == "@continuous":
+                    dag_dict["timetable"] = {
+                        "__var": {},
+                        "__type": "airflow.timetables.simple.ContinuousTimetable",
+                    }
+                elif sched == "@daily":
+                    dag_dict["timetable"] = {
+                        "__var": {
+                            "interval": 0.0,
+                            "timezone": "UTC",
+                            "expression": "0 0 * * *",
+                            "run_immediately": False,
+                        },
+                        "__type": "airflow.timetables.trigger.CronTriggerTimetable",
+                    }
+                else:
+                    # We should maybe convert this to None and warn instead
+                    raise ValueError(f"Unknown schedule_interval field {sched!r}")
+            elif sched.get("__type") == "timedelta":
+                dag_dict["timetable"] = {
+                    "__type": "airflow.timetables.interval.DeltaDataIntervalTimetable",
+                    "__var": {"delta": sched["__var"]},
+                }
+        elif timetable := dag_dict.get("timetable"):
+            if timetable["__type"] in {
+                "airflow.timetables.simple.DatasetTriggeredTimetable",
+                "airflow.timetables.datasets.DatasetOrTimeSchedule",
+            }:
+                dag_dict["timetable"] = _replace_dataset_with_asset_in_timetables(dag_dict["timetable"])
+
+        if "dag_dependencies" in dag_dict:
+            for dep in dag_dict["dag_dependencies"]:
+                for fld in ("dependency_type", "target", "source"):
+                    if dep.get(fld) == "dataset":
+                        dep[fld] = "asset"
+
+        for task in dag_dict["tasks"]:
+            task_var: dict = task["__var"]
+            if "airflow.ti_deps.deps.ready_to_reschedule.ReadyToRescheduleDep" in task_var.get("deps", []):
+                task_var["_is_sensor"] = True
+            for k in tasks_remove:
+                task_var.pop(k, None)
+            for old, new in task_renames:
+                task_var[new] = task_var.pop(old)
+            for item in task_var.get("outlets", []):
+                if isinstance(item, dict) and "__type" in item:
+                    item["__type"] = replace_dataset_in_str(item["__type"])
+            for item in task_var.get("inlets", []):
+                if isinstance(item, dict) and "__type" in item:
+                    item["__type"] = replace_dataset_in_str(item["__type"])
+                var_ = item["__var"]
+                var_["name"] = None
+                var_["group"] = None
+
+        # Set on the root TG
+        dag_dict["task_group"]["group_display_name"] = ""
```

Comment:
Probably should separate it out into a different file and have a better way of handling it later on -- but fine for now